tests/e2e/performance/test_small_file_workload.py

```diff
@@ -5,24 +5,316 @@
+    """
+    This class is reading all test results from elasticsearch server (which the
+    ripsaw running of the benchmark is generate), aggregate them by :
+        test operation (e.g. create / delete etc.)
+          sample (for test to be valid it need to run with more the one sample)
+            host (test can be run on more then one pod {called host})
+
+    it generate results for the all test as one unit which will be valid only
+    if the deviation between samples is less the 5%
+
+    """
+
+    managed_keys = {
+        'IOPS': {'name': 'iops', 'op': np.sum},
+        'MiBps': {'name': 'mbps', 'op': np.sum},
+        'elapsed': {'name': 'elapsed-time', 'op': np.average},
+        'files': {'name': 'Files-per-thread', 'op': np.sum},
+        'files-per-sec': {'name': 'Files-per-sec', 'op': np.sum},
+        'records': {'name': 'Rec-per-thread', 'op': np.sum},
+    }
+
+    def __init__(self, uuid, crd):
+        """
+        Initialize the object by reading some of the data from the CRD file and
+        by connecting to the ES server and read all results from it.
+
+        Args:
+            uuid (str): the unique uid of the test
+            crd (dict): dictionary with test parameters - the test yaml file
+                        that modify it in the test itself.
+
+        """
+
+        self.uuid = uuid
+
+        self.server = crd['spec']['elasticsearch']['server']
+        self.port = crd['spec']['elasticsearch']['port']
+        self.index = crd['spec']['es_index'] + '-results'
+        self.new_index = crd['spec']['es_index'] + '-fullres'
+        self.all_results = {}
+
+        # make sure we have connection to the elastic search server
+        log.info(f'Connecting to ES {self.server} on port {self.port}')
+        try:
+            self.es = Elasticsearch([{'host': self.server, 'port': self.port}])
+        except ESExp.ConnectionError:
+            log.error('can not connect to ES server {}:{}'.format(
+                self.server, self.port))
+            raise
+
+        # Creating full results dictionary
+        self.results = {
+            'clustername': crd['spec']['clustername'],
+            'clients': crd['spec']['workload']['args']['clients'],
+            'samples': crd['spec']['workload']['args']['samples'],
+            'threads': crd['spec']['workload']['args']['threads'],
+            'operations': crd['spec']['workload']['args']['operation'],
+            'uuid': uuid,
+            'full-res': {}
+        }
+
+        # Calculate the number of records for the test
+        self.records = self.results['clients'] * self.results['threads']
+        self.records *= self.results['samples']
+        self.records *= len(self.results['operations'])
+
+    def add_key(self, key, value):
+        """
+        Adding key and value the this object results dict
+
+        Args:
+            key (str): String which will be the key for the value
+            value (*): value to add, can be any kind of data type
```

Comment:
what is the type of value?

Reply:
As it said in the description : any kind of data type - str/list/dict/int/float

Reply:
done